### ISSUE TITLE
OOMKiller in default examples

### ISSUE THREAD

**Issue by rsff** [GoogleCloudPlatform/continuous-deployment-on-kubernetes#99]
Recently I started to had Jenkins crashing when the login was successful.
Following the logs showed OOMKiller being used, I tried to chase why a few weeks ago was working, the only major difference was the kubernetes cluster version but I'm guessing that would mean a regression, or it could also be the Jenkins issue.
I adjusted the values for memory limits/requests and after that everything worked fine.

Cheers,
RF

**Comment by jacobfederer** [comment]
I ran into the same issue. My guess is that the JVM doesn't respect the given boundaries which causes Kubernetes to panic into OOM.

I fixed it by setting the JAVA_OPTS environmental variable just slightly under the deployment Kubernetes restriction so the underlaying linux system has some memory available. E.g. if you have the default 1 gigabyte deployment constraint set 
```
- name: JAVA_OPTS
          value: '-Xms900m -Xmx900m'
```
